source/opt/compact_ids_pass.cpp

```diff
@@ -56,6 +56,17 @@ Pass::Status CompactIdsPass::Process() {
           }
           ++operand;
         }
+        if (inst->GetDebugScope().GetLexicalScope() != kNoDebugScope) {
+          auto it =
+              result_id_mapping.find(inst->GetDebugScope().GetLexicalScope());
+          if (it != result_id_mapping.end())
+            inst->UpdateLexicalScope(it->second);
```

Comment:
You should be setting modified to true when you change something.  You should only update the value if it is actually a different value.

Reply:
done